js/src/Zone.js

```diff
@@ -0,0 +1,57 @@
+import ExpectedMark from "./ExpectedMark";
+import UXCapture from "./UXCapture";
+
+// array holding current, global set of expected zones
+let expectedZones = [];
+
+/**
+ * Zone represents collection of elements groupped together and corresponding phase of page load
+ *
+ * Only one set of zones can be tracked for the same view at one point in time
```

Comment:
I think it might be useful to go ahead and introduce the concept of a 'view' into code, which would be the object that groups the 'expected zones'. I think that would make the logic more clear. My understanding is that:

- one View has many Zones (typically 4)
- one Zone may have many Marks, where the 'latest' recorded mark defines the end time of the measure. Currently, all zones treat 'navigationStart' as the 'start mark'

Reply:
I agree, it seems like a good abstraction to reflect a collection of zones, especially when we will be transitioning between SSR Views AKA "Page Views" AKA "SPA Hard" navigations and client-side Views AKA "Interactive Views" AKA "SPA Soft" navigations.

It's probably better to keep transition start point (e.g. navigationStart & start mark) on View rather than the zone and create metrics by defining measures between transition start and last event (UserTiming mark) in the zone.